torch/_dynamo/eval_frame.py

```diff
@@ -516,13 +580,17 @@ def catch_errors_wrapper(callback, hooks: Hooks):
     def catch_errors(frame, cache_entry, frame_state):
         assert frame_state is not None
 
+        is_skipfile = skipfiles.check(frame.f_code)
         if (
             # TODO: the first condition is not covered by any test
             frame.f_lasti >= first_real_inst_idx(frame.f_code)
-            or skipfiles.check(frame.f_code)
+            or is_skipfile
             or config.disable
         ):
-            log.debug("skipping %s %s", frame.f_code.co_name, frame.f_code.co_filename)
+            if not is_skipfile or config.verbose:  # don't log verbose skipfile skips
```

Comment:
@ezyang continuing from prev discussion this is verbose for the following:

Obtained by running `python -m pytest test/dynamo/test_logging.py -vv -s -k dynamo_debug`

But I think it's because it i

```
test/dynamo/test_logging.py::LoggingTests::test_dynamo_debug <- torch/testing/_internal/logging_utils.py [2023-10-14 16:42:43,872] torch._dynamo.eval_frame: [DEBUG] Saving dynamo config and hash for new compiled object(s). Hash: 9e9ed848cd0751349ffd46d2042c96c5
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping helper /usr/lib/python3.10/contextlib.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping __init__ /usr/lib/python3.10/contextlib.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping __enter__ /usr/lib/python3.10/contextlib.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping backend_cache_wrapper /home/jonch/Desktop/Programming/mlsys/pytorch/torch/_dynamo/eval_frame.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping _maybe_init_guarded_backend_cache /home/jonch/Desktop/Programming/mlsys/pytorch/torch/_dynamo/eval_frame.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping innermost_fn /home/jonch/Desktop/Programming/mlsys/pytorch/torch/_dynamo/eval_frame.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping _set_current_backend /home/jonch/Desktop/Programming/mlsys/pytorch/torch/_dynamo/eval_frame.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping __init__ /usr/lib/python3.10/contextlib.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping __enter__ /usr/lib/python3.10/contextlib.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping restore_guarded_dynamo_config /home/jonch/Desktop/Programming/mlsys/pytorch/torch/_dynamo/eval_frame.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping _maybe_init_guarded_config_cache /home/jonch/Desktop/Programming/mlsys/pytorch/torch/_dynamo/eval_frame.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping debug /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping isEnabledFor /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping _log /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping findCaller /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping <lambda> /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping normcase /usr/lib/python3.10/posixpath.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping makeRecord /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping __init__ /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping __instancecheck__ /usr/lib/python3.10/abc.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping getLevelName /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping basename /usr/lib/python3.10/posixpath.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping _get_sep /usr/lib/python3.10/posixpath.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] skipping splitext /usr/lib/python3.10/posixpath.py
[2023-10-14 16:42:43,873] torch._dynamo.convert_frame: [DEBUG] skipping because no torch.* _splitext             /usr/lib/python3.10/genericpath.py 121
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping current_thread /usr/lib/python3.10/threading.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping name /usr/lib/python3.10/threading.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping current_process /usr/lib/python3.10/multiprocessing/process.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping name /usr/lib/python3.10/multiprocessing/process.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping handle /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping filter /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping callHandlers /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping handle /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping acquire /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping new_emit /home/jonch/Desktop/Programming/mlsys/pytorch/torch/testing/_internal/logging_utils.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping emit /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping format /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping format /home/jonch/Desktop/Programming/mlsys/pytorch/torch/_logging/_internal.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping getLogger /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping getLogger /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping _acquireLock /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping _releaseLock /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping getMessage /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping formatTime /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping is_available /home/jonch/Desktop/Programming/mlsys/pytorch/torch/distributed/__init__.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping is_initialized /home/jonch/Desktop/Programming/mlsys/pytorch/torch/distributed/distributed_c10d.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping WORLD /home/jonch/Desktop/Programming/mlsys/pytorch/torch/distributed/distributed_c10d.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping default_pg /home/jonch/Desktop/Programming/mlsys/pytorch/torch/distributed/distributed_c10d.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping current_trace_id /home/jonch/Desktop/Programming/mlsys/pytorch/torch/_guards.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping get /home/jonch/Desktop/Programming/mlsys/pytorch/torch/_guards.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping <genexpr> /home/jonch/Desktop/Programming/mlsys/pytorch/torch/_logging/_internal.py
[2023-10-14 16:42:43,873] torch._dynamo.eval_frame: [DEBUG] Setting top-level compile config hash: 9e9ed848cd0751349ffd46d2042c96c5
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping flush /usr/lib/python3.10/logging/__init__.py
[2023-10-14 16:42:43,874] torch._dynamo.eval_frame: [DEBUG] skipping release /usr/lib/python3.10/logging/__init__.py
```

Which I do believe is too much when running on this simple function:
```
def example_fn(a):
    output = a.mul(torch.ones(1000, 1000))
    output = output.add(torch.ones(1000, 1000))
    return output
```

The reason seems to be tracing through nested `_TorchDynamoContext` context setup.

Reply:
I believe this would even trigger if we trace through `_DisableContext` etc.

Reply:
Following up to https://github.com/pytorch/pytorch/pull/111074#discussion_r1355976394

Reply:
@ezyang still need your advice on whether we should get rid of this verbose skipfiles logging.

Reply:
I'm just going to stack this to defer the changes, for now we will accept the verbosity.